searxng/searxng
# [BREAKING CHANGE] hostname replace plugin rewrite

## What does this PR do?
- refactors the hostname replace plugin to a `hostnames` plugin with additional features:
    - `high_priority` hostnames: these websites will get a higher score/ranking
    - `low_priority` hostnames: these websites will get a lower score/ranking
- basically an implementation of https://github.com/searxng/searxng/issues/1086#issuecomment-1097049011
- additionally this adds support for external list files as requested in #2619. Usage explanation: https://github.com/searxng/searxng/pull/3463/files#diff-e95e3f454925bc2344d3cf538950b11a1255935b47a609c9589e6ac33d74802fR246-R254

## How to test this PR locally?
- uncomment the `hostnames` configuration and enable the hostnames plugin
- search for `google`, `youtube`, `forgejo` (to see codeberg up in the results), ...

## Related issues

- closes #1086
- closes #2619
